src/nouislider.js

```diff
@@ -721,18 +812,20 @@
             entry = [entry, entry];
         }
 
-        // 'getMargin' returns false for invalid values.
-        parsed.padding = [parsed.spectrum.getMargin(entry[0]), parsed.spectrum.getMargin(entry[1])];
+        // 'getDistance' returns false for invalid values.
+        parsed.padding = [parsed.spectrum.getDistance(entry[0]), parsed.spectrum.getDistance(entry[1])];
 
-        if (parsed.padding[0] === false || parsed.padding[1] === false) {
-            throw new Error("noUiSlider (" + VERSION + "): 'padding' option is only supported on linear sliders.");
-        }
-
-        if (parsed.padding[0] < 0 || parsed.padding[1] < 0) {
-            throw new Error("noUiSlider (" + VERSION + "): 'padding' option must be a positive number(s).");
+        for (index = 0; index < parsed.spectrum.xNumSteps.length - 1; index++) {
+            // last "range" can't contain step size as it is purely an endpoint.
+            if (parsed.padding[0][index] < 0 || parsed.padding[1][index] < 0) {
+                throw new Error("noUiSlider (" + VERSION + "): 'padding' option must be a positive number(s).");
+            }
         }
 
-        if (parsed.padding[0] + parsed.padding[1] > 100) {
+        if (
+            (entry[0] + entry[1]) / (parsed.spectrum.xVal[parsed.spectrum.xVal.length - 1] - parsed.spectrum.xVal[0]) >
+            1
+        ) {
```

Comment:
I'm not sure about this. Would't adding the values in `parsed.padding` be enough?

Reply:
This goes back to my initial question how the getMargin function was developed, if I need to calculate it absolute or relative. The "old" getMargin calculates the percentage of distance in its current range. 

This makes it relative to it's current range. I've extended this whole code based upon that idea. It might be an idea to normalize the whole calculations and calculate the distance absolute (so over the whole range instead of the current). And normalize it so all factors are between 0 and 1. I must say: I like this idea but it is not a quick job to do as the whole code base must be normalized. And I am not sure of the impact of this.

Now back to your question: no this is not possible as the padding (and limit and margin) values are calculated relative to it's range. So a padding on the left of 100 in a range of 100 has a padding percentage of 100. A padding on the right of 100 in a range of 200 has a percentage of 50 (relative to it's range). So adding them up does not work anymore. 
The only way to calculate this again is with a getDistance. So if padding 1 + padding 2 in a getDistance if larger then 100. Net result will be the same then this.